docs: Fix unclear and partly wrong description of task dependencies (#27711)
gradle/gradle: platforms/documentation/docs/src/docs/userguide/reference/command_line_interface.adoc

.Simple Task Graph
image::commandLineTutorialTasks.png[]

> You can see that the `test` task is not executed, even though the `dist` task depends on it. The `test` task's dependencies, such as `compileTest`, are not executed either. Those dependencies of `test` that another task also depends upon, such as `compile`, are still executed.

Reviewer: ```suggestion
You can see that the `test` task is not executed, even though it is a dependency of the `dist` task. The `test` task’s dependencies, such as `compileTest`, are not executed either. 
The dependencies of `test` that other tasks depend on, such as `compile`, are still executed.
```
